Fix/stable latest (#1046)
PennyLaneAI/pennylane-lightning: .github/CHANGELOG.md

* Merge the `v0.40.0-rc` branch to the master and bump version.
  [(#1038)](https://github.com/PennyLaneAI/pennylane-lightning/pull/1038)

> * Reduce the number of shots on PennyLane Python tests, from 20k to 10k.

Reviewer: ```suggestion
* Reduce the number of shots in the PennyLane Python tests on CIs, from 20k to 10k.
```

  [(#1046)](https://github.com/PennyLaneAI/pennylane-lightning/pull/1046)

### Documentation

### Bug fixes

> * Patch `MultiControlledX` tests to fix stable/latest failures after the v0.40.0 release.

Reviewer 2: ```suggestion

* Patch `MultiControlledX` tests to fix stable/latest failures after the v0.40.0 release.
```

### Documentation

### Bug fixes

> * Patch MultiControlledX tests to fix current stable/latest fails.

Reviewer 2: ```suggestion

* Patch MultiControlledX tests to fix current stable/latest fails.
```

> * Patch MultiControlledX tests to fix current stable/latest fails.

Reviewer: ```suggestion
* Patch `MultiControlledX` tests to fix stable/latest failures after the v0.40.0 release.
```

> * Patch `MultiControlledX` tests to fix stable/latest failures after the v0.40.0 release.

Reviewer: @AmintorDusko Could you also add the shots reduction to the changelog for the future reference?
